agent/api/ecsclient/client.go

```diff
@@ -191,11 +201,40 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 
 	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
 	if err != nil {
-		log.Error("Could not register", "err", err)
+		log.Error("Could not register", "error", err)
 		return "", err
 	}
 	log.Info("Registered!")
-	return *resp.ContainerInstance.ContainerInstanceArn, nil
+	err = ValidateRegisteredAttributes(&registerRequest.Attributes, &resp.ContainerInstance.Attributes)
+	return *resp.ContainerInstance.ContainerInstanceArn, err
+}
+
+func attributesToMap(attributes *[]*ecs.Attribute) map[string]string {
+	attributeMap := make(map[string]string)
+	attribs := *attributes
+	for i := range attribs {
+		if attribs[i].Value != nil {
+			attributeMap[*attribs[i].Name] = *attribs[i].Value
+		} else {
+			attributeMap[*attribs[i].Name] = ""
+		}
+	}
+	return attributeMap
+}
+
+func ValidateRegisteredAttributes(reqAttributes, respAttributes *[]*ecs.Attribute) error {
```

Comment:
How about `expectedAttributes` and `actualAttributes` instead of `req` and `resp`?